x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,166 @@
+package keeper
+
+import (
+	"fmt"
+
+	ibcclienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	tmtypes "github.com/cometbft/cometbft/types"
+
+	"github.com/cosmos/interchain-security/v3/x/ccv/provider/types"
+)
+
+// HandleConsumerMisbehaviour checks if the given IBC misbehaviour corresponds to an equivocation light client attack,
+// and in this case, jails the Byzantine validators
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	logger := k.Logger(ctx)
+
+	// Check that the misbehaviour is valid and that the client consensus states at trusted heights are within trusting period
+	if err := k.CheckMisbehaviour(ctx, misbehaviour); err != nil {
```

Comment:
I would have expected that `CheckMisbehaviour` would check that singatures are valid based on some public keys, etc. under the hoods, similarly to what we do for double voting, but couldn't find something like this.

What guarantees us that we won't accept 2 arbitrary headers for the sam eheight with real validator addresses and arbitrary signatures that would lead to slashing?

Reply:
Solved in https://github.com/cosmos/interchain-security/pull/1401.